packages/grid/x-data-grid/src/internals/hooks/features/selection/useGridSelection.ts

```diff
@@ -45,19 +66,10 @@ export const useGridSelection = (
 ): void => {
   const logger = useGridLogger(apiRef, 'useGridSelection');
 
-  const propSelectionModel = React.useMemo(() => {
-    if (props.selectionModel == null) {
-      return props.selectionModel;
-    }
-
-    if (Array.isArray(props.selectionModel)) {
-      return props.selectionModel;
-    }
-
-    return [props.selectionModel];
-  }, [props.selectionModel]);
-
-  useGridStateInit(apiRef, (state) => ({ ...state, selection: propSelectionModel ?? [] }));
+  const propSelectionModel = React.useMemo(
```

Comment:
Small behavior change here that I think we can accept
If a user control and gives a non-array value, before I was not doing a new `setState` in the `useEffect` tracking the model.
Now it does because the array created in `getSelectionModelPropValue` called to initialize the state won't have the same reference and the array called to create `propSelectionModel`.